### PR TITLE
Refactor SparqlQueryView

### LINKED ISSUE #316: Refactor SparqlQueryView
implements #300

### PR DESCRIPTION
**Pull request by kenjinp**
fixes #316

### CODE REVIEW

File: src/shared/components/ViewForm/SparqlQueryForm.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { Form, Button, Card, List, Empty, Table, Tooltip } from 'antd';
+import Column from 'antd/lib/table/Column';
+import { UnControlled as CodeMirror } from 'react-codemirror2';
+import * as hash from 'object-hash';
+
+import 'codemirror/addon/display/placeholder';
+import 'codemirror/mode/sparql/sparql';
+
+// TODO move once SparqlQueryView is also refactored
```
[the remark this answers is not in the document]
Comment (kenjinp) [reply src/shared/components/ViewForm/SparqlQueryForm.tsx]: I forgot to remove this TODO when I did the thing

[the remark this answers is not in the document]
Comment (julienmachon) [reply src/shared/components/ViewForm/SparqlQueryForm.tsx]: My todo extension only works with `// TODO: blabla`, not with `// TODO blabla`

File: src/shared/components/ViewForm/SparqlQueryForm.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  type: string;
+};
+
+type Bindings = {
+  [variableName: string]: Entry;
+}[];
+
+const SparqlQueryForm: React.FunctionComponent<{
+  query: string;
+  // TODO update response type
```
Comment (julienmachon) [inline src/shared/components/ViewForm/SparqlQueryForm.tsx]:
```suggestion
  // TODO: update response type
```

File: src/shared/views/SparqlQueryView.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  ): void;
+}> = ({ match, location, goToOrg, goToProject, goToView }): JSX.Element => {
+  const {
+    params: { org: orgLabel, project: projectLabel, viewId },
+  } = match;
+  const [{ _results: views }, setViews] = React.useState<ViewList>({
+    '@context': {},
+    total: 0,
+
+    // @ts-ignore TODO fix incorrect typing in SDK
```
Comment (julienmachon) [inline src/shared/views/SparqlQueryView.tsx]:
```suggestion
    // @ts-ignore TODO: fix incorrect typing in SDK
```

File: src/shared/views/SparqlQueryView.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  const nexus = useNexusContext();
+  const decodedViewId = decodeURIComponent(viewId);
+  // const targetView = views.find((view: View) => view['@id'] === decodedViewId);
+  const query = queryString.parse(location.search).query;
+
+  React.useEffect(() => {
+    nexus.View.list(orgLabel, projectLabel)
+      .then(setViews)
+      .catch(() => {
+        // 503 ?
```
Comment (julienmachon) [inline src/shared/views/SparqlQueryView.tsx]: can we like, display an error message in the menu (as a disabled item) saying something went wrong while fetching the list of views?

**Pull request merged**